test/manyvariables.jl

```diff
@@ -133,6 +133,8 @@ end
     @test HomogeneousPolynomial(0,0)  == 0
     @test (@inferred conj(xH)) == (@inferred adjoint(xH))
     @test (@inferred real(xH)) == xH
+    @test xH > yH
```

Comment:
It should be a lot more obvious from the test what it is that we are trying to test and why this is the expected result.

Reply:
The test checks the *lexicographic* part in the ordering for `HomogeneousPolynomial`s, here comparing same orders; other tests (below) do something alike for different orders.